packages/banner/src/BannerAnimator/BannerAnimator.js

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { polyfill } from "react-lifecycles-compat";
+
+import { positions, AVAILABLE_POSITIONS } from "./positions";
+import statuses from "./statuses";
+
+import {
+  animateCollapse,
+  animateExpand,
+  endCollapse,
+  endExpand,
+  prepareCollapse,
+  startCollapse,
+  startExpand
+} from "./updaters";
+
+/**
+ * @typedef {Object} BannerAnimatorProps
+ * @property {boolean} [isVisible]
+ * @property {boolean} [hasBounce]
+ * @property {boolean} [hasPush]
+ * @property {string} [position]
+ * @property {function(ContainerBag): JSX.Element} children
+ */
+
+/**
+ * @typedef {Object} BannerAnimatorState
+ * @property {string} [status]
+ * @property {Object.<string, string>} [wrapperStyle]
+ * @property {Object.<string, string>} [innerWrapperStyle]
+ * @property {HTMLDivElement} [wrapper]
+ * @property {HTMLDivElement} [innerWrapper]
+ */
+
+/** @type {Component<BannerAnimatorProps, BannerAnimatorState>} */
+class BannerAnimator extends Component {
+  /**
+   * @param {BannerAnimatorProps} nextProps
+   * @param {BannerAnimatorState} prevState
+   * @returns {BannerAnimatorState | null}
+   */
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { isVisible } = nextProps;
+    const { status } = prevState;
+
+    switch (status) {
+      case statuses.COLLAPSED:
+      case statuses.COLLAPSING:
+        return isVisible ? startExpand() : null;
+      case statuses.EXPANDED:
+      case statuses.EXPANDING:
+        return isVisible ? null : startCollapse();
+      default:
+        return isVisible ? endExpand() : endCollapse();
```

Comment:
Leave `default` for unexpected values. Move this above and check for falsy values specially.